src/sentry/killswitches.py

```diff
@@ -0,0 +1,120 @@
+type and organization ID.
+
+This is similar to existing featureflagging systems we have, but with less
+features and more performant.
+"""
+
+import copy
+from collections import namedtuple
+from typing import Any, Dict, List, Union
+
+from sentry import options
+from sentry.utils import metrics
+
+Condition = Dict[str, str]
+KillswitchConfig = List[Condition]
+LegacyKillswitchConfig = Union[KillswitchConfig, List[int]]
+Context = Dict[str, Any]
+
+KillswitchInfo = namedtuple("KillswitchInfo", ["description", "fields"])
+
+ALL_KILLSWITCH_OPTIONS = {
+    "store.load-shed-group-creation-projects": KillswitchInfo(
+        description="Drop event in save_event before entering transaction to create group",
+        fields=("project_id", "platform"),
+    ),
+    "store.load-shed-pipeline-projects": KillswitchInfo(
+        description="Drop event in ingest consumer. Available fields are severely restricted because nothing is parsed yet.",
+        fields=("project_id", "event_id", "has_attachments"),
+    ),
+    "store.load-shed-parsed-pipeline-projects": KillswitchInfo(
+        description="Drop events in ingest consumer after parsing them. Available fields are more but a bunch of stuff can go wrong before that.",
+        fields=("organization_id", "project_id", "event_type", "has_attachments", "event_id"),
+    ),
+    "store.load-shed-process-event-projects": KillswitchInfo(
+        description="Drop events in process_event.",
+        fields=("project_id", "event_id", "platform"),
+    ),
+    "store.load-shed-symbolicate-event-projects": KillswitchInfo(
+        description="Drop events in symbolicate_event.",
+        fields=("project_id", "event_id", "platform"),
+    ),
+}
+
+
+def normalize_value(option_value: Any) -> KillswitchConfig:
+    rv = []
+    for condition in option_value:
+        if not condition:
+            continue
+        elif isinstance(condition, int):
+            rv.append({"project_id": str(condition)})
+        elif isinstance(condition, dict):
+            condition = {k: str(v) for k, v in condition.items() if v is not None}
+            if condition:
+                rv.append(condition)
+
+    return rv
+
+
+def killswitch_matches_context(option_key: str, context: Context) -> bool:
+    assert option_key in ALL_KILLSWITCH_OPTIONS
+    assert set(ALL_KILLSWITCH_OPTIONS[option_key].fields) == set(context)
+    option_value = options.get(option_key)
+    rv = _value_matches(option_value, context)
+    metrics.incr(
+        "sentry.killswitches.run",
+        tags={"option_key": option_key, "decision": "matched" if rv else "passed"},
+    )
+
+    return rv
+
+
+def _value_matches(raw_option_value: LegacyKillswitchConfig, context: Context) -> bool:
+    option_value = normalize_value(raw_option_value)
+
+    for condition in option_value:
+        for field, matching_value in condition.items():
+            value = context.get(field)
+            if value is None:
```

Comment:
Why is `None` not allowed as a value for a context field?

Reply:
I think it would be too confusing if `null` is not the same as absence